snuba/settings/validation.py

```diff
@@ -55,7 +55,8 @@ def _validate_settings(locals: Mapping[str, Any]) -> None:
 
     for cluster in locals["CLUSTERS"]:
         for cluster_storage_set in cluster["storage_sets"]:
-            storage_set_to_cluster[StorageSetKey(cluster_storage_set)] = cluster
+            if cluster_storage_set in StorageSetKey.to_set():
+                storage_set_to_cluster[StorageSetKey(cluster_storage_set)] = cluster
```

Comment:
```suggestion
try:
    storage_set_to_cluster[StorageSetKey(cluster_storage_set)] = cluster
except ValueError:
    pass
```
You could remove StorageSetKey.to_set